tests/compatibility-testing/programs/compatibility-testing/tests/solana_program_test.rs

```diff
@@ -0,0 +1,10 @@
+use solana_program_test::{processor, ProgramTest};
+
+#[test]
+fn entrypoint_lifetime() {
+    let _pt = ProgramTest::new(
+        "compatibility_testing",
+        compatibility_testing::id(),
+        processor!(compatibility_testing::entry),
```

Comment:
I think people started passing `None` to the `builtin_function` argument, and let `cargo test-sbf` load the programs:

https://github.com/anza-xyz/agave/blob/a2579d484efd78099ba67db5911b2ba12c77b711/program-test/src/lib.rs#L617-L621

That way, the incompatibility of the lifetimes is not an issue and `solana-program-test` can be used with the generated `entry` function.

Reply:
Thanks for the context! Done!